src/wp-includes/class-walker-nav-menu.php

```diff
@@ -276,4 +307,24 @@ public function end_el( &$output, $data_object, $depth = 0, $args = null ) {
 		$output .= "</li>{$n}";
 	}
 
+	/**
+	 * Builds an HTML attributes string based on an array of key value pairs
+	 * Empty values are ignored
+	 *
+	 * @since 6.1.2
+	 *
+	 * @param      array   $atts   The atts
+	 * @return     string  HTML attributes string
+	 */
+	protected function build_atts( $atts = array() ) {
+		$attributes = '';
+		foreach ( $atts as $attr => $value ) {
+			if ( is_scalar( $value ) && '' !== $value && false !== $value ) {
```

Comment:
```suggestion
			if ( false !== $value && is_scalar( $value ) && '' !== trim( $value ) ) {
```
1. Slight re-ordering to skip early if `$value` is `false`.
2. `trim( $value )` to prevent values with only blank space. - If this isn't preferred here, let me know and include an example of an acceptable empty value to help clarify this for me.

Reply:
I agree on the reordering, but blank values are allowed in HTML values so trimming shouldn't be added here. The main thing that I would like to further address in a future patch would be to allow empty html attributes (ex.: "disabled" without any value or <span aria-hidden>Something</span>) This would involve a bit of refactoring. I'll resolve this for now.